cdap-common/src/main/java/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java

```diff
@@ -126,22 +141,27 @@ private Resource adjustMemory(GetNewApplicationResponse response, Resource capab
 
   @Override
   public ProcessController<YarnApplicationReport> createProcessController(ApplicationId appId) {
-    return new ProcessControllerImpl(yarnClient, appId);
+    return new ProcessControllerImpl(createYarnClient(), appId);
```

Comment:
It's passed into a `ProcessControllerImpl`, and it'll get closed when the ProcessController#cancel gets called by the caller of this method.